repository/spark/docs/latest/security.md

```diff
@@ -0,0 +1,174 @@
+Security
+---
+
+### Securing Spark RPC communication
+
+Spark supports authentication for RPC channels (protocol between Spark processes), which allows secure 
+communication between driver and executors and also adds a possibility to enable network encryption between Spark processes.
+For more information refer to the [official Spark documentation](https://spark.apache.org/docs/latest/security.html#encryption).
+
+#### Authentication and encryption
+To enable authentication for RPC, the following configuration is required:
+```
+# enable authentication for internal connections
+spark.authenticate  true  
+
+# the secret key, used for authentication. Must be configured on each of the nodes.          
+spark.authenticate.secret  spark-secret
+```
+If you want to set up the encryption for RPC connections, you can enable it by setting `spark.network.crypto.enabled` to `true`.
+Spark authentication must be enabled for encryption to work.
+Additional configuration properties can be found in Spark documentation.
+
+The example below describes how to set up authentication and encryption for `SparkApplication` on Kubernetes.
+Also, it shows how to provide a custom `log4j.properties` to change the logging strategy.
```

Comment:
I think that logging example deserves a separate section in more generic configuration docs - it looks a bit confusing in the security section.

Reply:
moved to a separate section under `Submitting Spark Applications`